chore!: Remove deprecated content (#5562)
SAP/cloud-sdk-js: V4-Upgrade-Guide.md

- [Update Your Project Dependencies](#update-your-project-dependencies)
- [Update to Node 22 or Newer](#update-to-node-22-or-newer)
- [Set `useCache` explicitly to false to turn off destination caching](#set-usecache-explicitly-to-false-to-turn-off-destination-caching)

> - [Check for Deprecation](#check-for-deprecation)

Reviewer: [q] I feel deprecation here can be misleading and now its not deprecated functionality, rather 'removed functionality'

Author: I changed it into `Remove Deprecated Content`.

.execute({ destinationName: 'DESTINATION', jwt: 'JWT', useCache: false })
```

## Check for Deprecation

- `@sap-cloud-sdk/connectivity`

>   - The behaviour of `getAgentConfig()` function is changed to be asynchronous. The temporary asynchroneous function `getAgentConfigAsync()` has been removed.

Reviewer: ```suggestion
  - The `getAgentConfig()` function is now asynchronous. The `getAgentConfigAsync()` function has been removed.
```
